backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropCallConvertors.kt

```diff
@@ -121,20 +88,12 @@ private fun InteropCallContext.writeValueToMemory(
     }
 }
 
-private fun InteropCallContext.determineInMemoryType(type: IrType): IrType {
-    val classifier = type.classOrNull!!
-    return if (classifier in symbols.unsignedIntegerClasses) {
-        symbols.unsignedToSignedOfSameBitWidth.getValue(classifier).owner.defaultType
-    } else {
-        type
-    }
-}
-
 private fun InteropCallContext.castPrimitiveIfNeeded(
         value: IrExpression,
         targetType: IrType
 ): IrExpression {
     val valueClass = value.type.classOrNull!!
+    if (valueClass === symbols.nothing) return value
```

Comment:
Same here:
```c
struct S {
    int x;
};
```

```kotlin
fun main() {
    foo(42)
}

fun <T : Int> foo(x: T) {
    val s = nativeHeap.alloc<S>()
    s.x = x
}
```

-> NPE above.

Please ensure you don't rely on type of IR expressions in this file.